lib/matplotlib/backend_managers.py

```diff
@@ -14,6 +18,197 @@
+        The figure number.
+    """
+    def __init__(self, figure, num, **kwargs):
+        super(FigureManager, self).__init__(**kwargs)
+        self._backend_name, self._backend = get_backend()
+
+        self.num = num
+        self.figure = figure
+
+        self._is_gui = hasattr(self._backend, 'Window')
+        if not self._is_gui:
+            self.window = None
+            return
+
+        self._mainloop = self._backend.MainLoop()
+        self.window = self._backend.Window('Figure %d' % num)
+        self.window.mpl_connect('window_destroy_event', self.destroy)
+
+        w = int(self.figure.bbox.width)
+        h = int(self.figure.bbox.height)
+
+        self.window.add_element(self.figure.canvas, 'center')
+
+        self.toolmanager = ToolManager(self.figure.canvas)
+        self.toolbar = self._get_toolbar()
+
+        tools.add_tools_to_manager(self.toolmanager)
+        if self.toolbar:
+            tools.add_tools_to_container(self.toolbar)
+            self.statusbar = self._backend.Statusbar(self.toolmanager)
+            h += self.window.add_element(self.statusbar, 'south')
+
+        if self.toolbar is not None:
+            h += self.window.add_element(self.toolbar, 'south')
+
+        self.window.set_default_size(w, h)
+        self._full_screen_flag = False
+
+        if is_interactive():
+            self.window.show()
+
+        def notify_axes_change(fig):
+            'this will be called whenever the current axes is changed'
+            if self.toolmanager is None and self.toolbar is not None:
+                self.toolbar.update()
+        self.figure.add_axobserver(notify_axes_change)
+
+    @property
+    def figure(self):
+        return self._figure
+
+    @figure.setter
+    def figure(self, figure):
+        if hasattr(self, '_figure'):
+            raise NotImplementedError
+
+        if not figure.canvas:
+            self._backend.FigureCanvas(figure, manager=self,
+                                       backend=self.backend)
+        self._figure = figure
+
+    @property
+    def canvas(self):
+        return self._figure.canvas
+
+    def destroy(self, *args):
+        """Called to destroy this FigureManager.
+        """
+
+        # Make sure we run this routine only once for the FigureManager
+        # This ensures the nasty __del__ fix below works.
+        if getattr(self, '_destroying', False) or self._is_gui is False:
+            return
+
+        self._destroying = True
+        self.figure.canvas.destroy()
+        if self.toolbar:
+            self.toolbar.destroy()
+        self.window.destroy()
+
+        # Fix as for some reason we have extra references to this#
+        # i.e. ``del self._mainloop`` doesn't work
+        self._mainloop.__del__()
+
+        s = 'window_destroy_event'
+        event = FigureManagerEvent(s, self)
+        self._callbacks.process(s, event)
+
+    def show(self):
+        """Shows the figure"""
+        self.window.show()
+        self.canvas.focus()
```

Comment:
Ahh, here, ``window.show()`` raises the window, ``self.canvas.focus()``, gives the canvas focus.

Yes, thinking back I think I did pause before putting this in, the reason, to ensure that the canvas initially starts with focus... what do you think about an if statement to detect if the window has already been shown?